odoo/tools/sql.py

```diff
@@ -196,6 +197,14 @@ def create_index(cr, indexname, tablename, expressions):
     cr.execute('CREATE INDEX "{}" ON "{}" ({})'.format(indexname, tablename, args))
     _schema.debug("Table %r: created index %r (%s)", tablename, indexname, args)
 
+def create_json_translation_index(cr, indexname, tablename, colname, lang):
+    """ Create the given index unless it exists. """
+    if index_exists(cr, indexname):
+        return
+    cr.execute('CREATE EXTENSION IF NOT EXISTS pg_trgm')
+    cr.execute('CREATE INDEX "{}" ON "{}" using GIN( ({}->>\'{}\') gin_trgm_ops)'.format(indexname, tablename, colname, lang))
```

Comment:
Currently, trigram indexes are considered as too expensive and should be optional

Reply:
@lmignon did you measure index size in your tests?